lib/extendedkey.ex

```diff
@@ -433,12 +440,12 @@ defmodule Bitcoinex.ExtendedKey do
   @spec derive_public_child(t(), non_neg_integer) :: t()
   def derive_public_child(xkey, idx) do
     cond do
-      xkey.prefix in prv_prefixes() ->
+      xkey.prefix in @prv_prefixes ->
         xkey
         |> derive_private_child(idx)
         |> to_extended_public_key()
 
-      idx >= @softcap or idx < 0 ->
+      idx >= DerivationPath.max_non_hardened_child_num() or idx < 0 ->
```

Comment:
Here, @softcap = 2^31 and max_non_hardened_child_num() = 2^31 -1.

Reply:
fixed 🙏🏽